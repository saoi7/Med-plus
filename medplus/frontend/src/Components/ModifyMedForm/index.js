import React from 'react';
import Input from '../Input';
import SubmitInput from '../SubmitInput';
import { ImPlus, ImMinus } from 'react-icons/im';
import * as ROUTES from '../../constants/routes';
import NavBar from '../NavBar';
import { withFirebase } from '../Firebase';
import { withRouter } from 'react-router-dom';

// TODO:
// add validation for:
// quantity must be > 0
// end date must come after start date
// name must not be empty

const MAX_TIME_TO_TAKE_INPUTS = 3;

class ModifyMedFormBase extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            edit_page_flag: false,
            submit_button_text: "",
            med_name: "",
            start_date: "",
            end_date: "",
            times_to_take: [
                { time_to_take: "", quantity: null }
            ],
        };
    }

    componentDidMount() {
        this.setState({
            ...this.props.initialState
        });
    }

    onSubmit = event => {
        event.preventDefault();

        const med_name = this.state.med_name;
        let modified_times_to_take = {};
        this.state.times_to_take.forEach((time_obj, index) => {
            let time_str = time_obj.time_to_take;
            let qty = time_obj.quantity;
            modified_times_to_take[time_str] = qty;
        });

        this.props.firebase.TEST_schedules(med_name).set({
            ...this.state,
            times_to_take: modified_times_to_take
        });

        // redirect to manage meds page
        this.props.history.push(ROUTES.MANAGE_MEDS);
    };

    handleTimeToTakeChange = index => event => {
        let new_times_to_take = [ ...this.state.times_to_take ];
        new_times_to_take[index] = { 
            ...this.state.times_to_take[index], 
            time_to_take: event.target.value
        };
        this.setState({
            times_to_take: new_times_to_take
        });
    }

    handleQuantityChange = index => event => {
        let new_times_to_take = [ ...this.state.times_to_take ];
        new_times_to_take[index] = { 
            ...this.state.times_to_take[index],
            quantity: event.target.value
        };
        this.setState({
            times_to_take: new_times_to_take
        });
    }

    onChange = event => {
        this.setState({ [event.target.name]: event.target.value });
    };

    addTimeToTakeQuantityGroup = event => {
        let new_times_to_take = [
            ...this.state.times_to_take,
            { time_to_take: "", quantity: null }
        ];
        this.setState({
            times_to_take: new_times_to_take
        });
    }

    removeTimeToTakeQuantityGroup = index => event => {
        let new_times_to_take = [
            ...this.state.times_to_take.slice(0, index),
            ...this.state.times_to_take.slice(index+1),
        ];
        this.setState({
            times_to_take: new_times_to_take
        });
    }

    render() {
        let { start_date, end_date, time_to_take, quantity, med_name } = this.state;
        const on_change_name_input = (this.state.edit_page_flag) ? e => e.preventDefault() : this.onChange;
        let times_to_take_entries = this.state.times_to_take.map((times_to_take_obj, index) => {
            let add_new_time_input_button = (
                <ImMinus className="time-to-take-button background-blue-hover" onClick={this.removeTimeToTakeQuantityGroup(index)} />
            );
            if(index === this.state.times_to_take.length -1 && this.state.times_to_take.length !== MAX_TIME_TO_TAKE_INPUTS) {
                add_new_time_input_button = (
                    <ImPlus className="time-to-take-button background-blue-hover" onClick={this.addTimeToTakeQuantityGroup} data-testid="add-time-to-take-button" />
                );
            }
            return (
                <div className="med-list-time-to-take-quantity-group">
                    <Input labelText="Time To Take" type="time" name={"time_to_take"} value={times_to_take_obj.time_to_take} onChange={this.handleTimeToTakeChange(index)} required />
                    <Input className="quantity-input" labelText="Quantity" type="number" name={"quantity"} value={times_to_take_obj.quantity} onChange={this.handleQuantityChange(index)} required />
                    { add_new_time_input_button }
                </div>
            );
        });

        let name_input = ( <Input labelText="Name" type="text" name="med_name" value={med_name} onChange={on_change_name_input} required /> );
        if(this.state.edit_page_flag) {
            name_input = null;
        }

        return (
            <form className="add-med-form background-blue" onSubmit={this.onSubmit} >
                { name_input }
                <Input labelText="Start Date" type="date" name="start_date" value={start_date} onChange={this.onChange} required />
                <Input labelText="End Date" type="date" name="end_date" value={end_date} onChange={this.onChange} required />
                { times_to_take_entries }
                <SubmitInput labelText={this.state.submit_button_text} />
            </form>
        );
    }
}

//const ModifyMedForm = withRouter(withFirebase(ModifyMedFormBase));
const ModifyMedForm = withRouter(ModifyMedFormBase);
export default ModifyMedForm;

export { ModifyMedFormBase };
export { MAX_TIME_TO_TAKE_INPUTS };
