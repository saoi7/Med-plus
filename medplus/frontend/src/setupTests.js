// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import '@testing-library/jest-dom';
import { unmountComponentAtNode } from "react-dom";

let test_dom_container = null;
beforeEach(() => {
  // setup a DOM element as a render target
  test_dom_container = document.createElement("div");
  document.body.appendChild(test_dom_container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(test_dom_container);
  test_dom_container.remove();
  test_dom_container = null;
});

export { test_dom_container };
