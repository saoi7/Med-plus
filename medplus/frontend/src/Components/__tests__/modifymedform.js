import { ModifyMedFormBase, MAX_TIME_TO_TAKE_INPUTS } from '../ModifyMedForm';
import {test_dom_container} from '../../setupTests';
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import renderer from 'react-test-renderer';
import { getByTestId, fireEvent } from '@testing-library/react';
import { MOCK_FIREBASE_DB_REF_PROP } from './mock_firebase_props';

// NOTE: you must provide an array as a 'history' prop
// this normally comes from react router so we must provide a
// dummy object here 

test("check no unintended changes have been made to rendered component", () => {
    const COMPONENT_NO_PROPS = renderer.create(
        <ModifyMedFormBase history={[]}/>
    );
    let tree = COMPONENT_NO_PROPS.toJSON();
    expect(tree).toMatchSnapshot();

    const STATE_1 = {
        edit_page_flag: false,
        page_title: "Add new medication",
        submit_button_text: "Add medication",
        med_name: "abc",
        start_date: "2021-10-20",
        end_date: "2021-10-30",
        times_to_take: [
            { time_to_take: "12:34", quantity: "2" },
            { time_to_take: "01:00", quantity: "1" },
        ],
    };
    const COMPONENT_WITH_PROPS_1 = renderer.create(
        <ModifyMedFormBase initialState={STATE_1} history={[]}/>
    );
    tree = COMPONENT_WITH_PROPS_1.toJSON();
    expect(tree).toMatchSnapshot();

    const STATE_2 = {
        edit_page_flag: true,
        page_title: "edit medication",
        submit_button_text: "edit",
        med_name: "2hwqlkmgtnqpewjirn",
        start_date: "2021-11-20",
        end_date: "2021-11-29",
        times_to_take: [
            { time_to_take: "12:34", quantity: "1" },
            { time_to_take: "03:00", quantity: "2" },
            { time_to_take: "04:00", quantity: "3" },
            { time_to_take: "05:00", quantity: "123" },
        ],
    };
    const COMPONENT_WITH_PROPS_2 = renderer.create(
        <ModifyMedFormBase initialState={STATE_2} history={[]}/>
    );
    tree = COMPONENT_WITH_PROPS_2.toJSON();
    expect(tree).toMatchSnapshot();

});

const verify_time_to_take_buttons_count = (initial_input_count) => {
    let ADD_TIME_TO_TAKE_BUTTON;
    for(let current_input_count = initial_input_count; current_input_count < MAX_TIME_TO_TAKE_INPUTS; current_input_count++)
    {
        ADD_TIME_TO_TAKE_BUTTON = getByTestId(test_dom_container, 'add-time-to-take-button');
        expect(ADD_TIME_TO_TAKE_BUTTON).toBeDefined();
        expect(ADD_TIME_TO_TAKE_BUTTON).not.toBeNull();
        act(() => {
            ADD_TIME_TO_TAKE_BUTTON.dispatchEvent(new MouseEvent('click', { bubbles: true }));
        });
    }

    const TIME_TO_TAKE_QUERY = document.querySelector("[data-testid=add-time-to-take-button]");
    expect(TIME_TO_TAKE_QUERY).toBeNull();

    const TIME_TO_TAKE_GROUP = document.getElementsByClassName('med-list-time-to-take-quantity-group');
    expect(TIME_TO_TAKE_GROUP.length).toBe(MAX_TIME_TO_TAKE_INPUTS);
}

test("check max time to take input count is correct with no initialState input", () => {
    act(() => {
        render(<ModifyMedFormBase history={[]}/>, test_dom_container);
    });
    // NOTE: the input count starts at 1 if no time_to_take inputs are set as props to ModifyMedForm
    verify_time_to_take_buttons_count(1);
});

test("check max time to take input count is correctwith an initialState input", () => {
    const initial_state = {
        edit_page_flag: false,
        page_title: "Add new medication",
        submit_button_text: "Add medication",
        med_name: "",
        start_date: "",
        end_date: "",
        times_to_take: [
            { time_to_take: "", quantity: null },
            { time_to_take: "", quantity: null },
        ],
    };
    act(() => {
        render(<ModifyMedFormBase initialState={initial_state} history={[]}/>, test_dom_container);
    });
    verify_time_to_take_buttons_count(initial_state.times_to_take.length);
});

test("check that name input can be modified when edit flag is not set", () => {
    const initial_state = {
        edit_page_flag: false,
        page_title: "Add new medication",
        submit_button_text: "Add medication",
        med_name: "",
        start_date: "",
        end_date: "",
        times_to_take: [
            { time_to_take: "", quantity: null },
        ],
    };
    act(() => {
        render(<ModifyMedFormBase initialState={initial_state} history={[]}/>, test_dom_container);
    });
    const TEST_INPUT_STRING = 'abcefgh';
    let MED_NAME_INPUT = document.querySelector('[name=med_name]');
    expect(MED_NAME_INPUT.value).toBe("");
    fireEvent.change(MED_NAME_INPUT, {target: {value: TEST_INPUT_STRING}});
    expect(MED_NAME_INPUT.value).toBe(TEST_INPUT_STRING);
});

test("check that name input cannot be modified when edit flag is set", () => {
    const initial_state = {
        edit_page_flag: true,
        page_title: "",
        submit_button_text: "",
        med_name: "",
        start_date: "",
        end_date: "",
        times_to_take: [
            { time_to_take: "", quantity: null },
        ],
    };
    act(() => {
        render(<ModifyMedFormBase initialState={initial_state} history={[]}/>, test_dom_container);
    });
    const TEST_INPUT_STRING = 'abcefgh';
    let MED_NAME_INPUT = document.querySelector('[name=med_name]');
    expect(MED_NAME_INPUT.value).toBe("");
    fireEvent.change(MED_NAME_INPUT, {target: {value: TEST_INPUT_STRING}});
    expect(MED_NAME_INPUT.value).toBe("");
});
