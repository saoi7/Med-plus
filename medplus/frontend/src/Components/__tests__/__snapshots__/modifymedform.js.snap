// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check no unintended changes have been made to rendered component 1`] = `
<form
  className="add-med-form"
  onSubmit={[Function]}
>
  <div
    className="add-med-input font-small undefined"
  >
    <label
      className="left-padding"
    >
      Name: 
    </label>
    <input
      className="hidden-text-box left-padding font-small"
      name="med_name"
      onChange={[Function]}
      required={true}
      type="text"
      value=""
    />
  </div>
  <div
    className="add-med-input font-small undefined"
  >
    <label
      className="left-padding"
    >
      Start Date: 
    </label>
    <input
      className="hidden-text-box left-padding font-small"
      name="start_date"
      onChange={[Function]}
      required={true}
      type="date"
      value=""
    />
  </div>
  <div
    className="add-med-input font-small undefined"
  >
    <label
      className="left-padding"
    >
      End Date: 
    </label>
    <input
      className="hidden-text-box left-padding font-small"
      name="end_date"
      onChange={[Function]}
      required={true}
      type="date"
      value=""
    />
  </div>
  <div
    className="med-list-time-to-take-quantity-group"
  >
    <div
      className="add-med-input font-small undefined"
    >
      <label
        className="left-padding"
      >
        Time To Take: 
      </label>
      <input
        className="hidden-text-box left-padding font-small"
        name="time_to_take"
        onChange={[Function]}
        required={true}
        type="time"
        value=""
      />
    </div>
    <div
      className="add-med-input font-small quantity-input"
    >
      <label
        className="left-padding"
      >
        Quantity: 
      </label>
      <input
        className="hidden-text-box left-padding font-small"
        name="quantity"
        onChange={[Function]}
        required={true}
        type="number"
        value={null}
      />
    </div>
    <svg
      className="time-to-take-button"
      data-testid="add-time-to-take-button"
      fill="currentColor"
      height="1em"
      onClick={[Function]}
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      version="1.1"
      viewBox="0 0 16 16"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M15.5 6h-5.5v-5.5c0-0.276-0.224-0.5-0.5-0.5h-3c-0.276 0-0.5 0.224-0.5 0.5v5.5h-5.5c-0.276 0-0.5 0.224-0.5 0.5v3c0 0.276 0.224 0.5 0.5 0.5h5.5v5.5c0 0.276 0.224 0.5 0.5 0.5h3c0.276 0 0.5-0.224 0.5-0.5v-5.5h5.5c0.276 0 0.5-0.224 0.5-0.5v-3c0-0.276-0.224-0.5-0.5-0.5z"
      />
    </svg>
  </div>
  <div
    className="add-med-input font-small"
  >
    <input
      className="hidden-submit-button font-small"
      type="submit"
      value=""
    />
  </div>
</form>
`;

exports[`check no unintended changes have been made to rendered component 2`] = `
<form
  className="add-med-form"
  onSubmit={[Function]}
>
  <div
    className="add-med-input font-small undefined"
  >
    <label
      className="left-padding"
    >
      Name: 
    </label>
    <input
      className="hidden-text-box left-padding font-small"
      name="med_name"
      onChange={[Function]}
      required={true}
      type="text"
      value="abc"
    />
  </div>
  <div
    className="add-med-input font-small undefined"
  >
    <label
      className="left-padding"
    >
      Start Date: 
    </label>
    <input
      className="hidden-text-box left-padding font-small"
      name="start_date"
      onChange={[Function]}
      required={true}
      type="date"
      value="2021-10-20"
    />
  </div>
  <div
    className="add-med-input font-small undefined"
  >
    <label
      className="left-padding"
    >
      End Date: 
    </label>
    <input
      className="hidden-text-box left-padding font-small"
      name="end_date"
      onChange={[Function]}
      required={true}
      type="date"
      value="2021-10-30"
    />
  </div>
  <div
    className="med-list-time-to-take-quantity-group"
  >
    <div
      className="add-med-input font-small undefined"
    >
      <label
        className="left-padding"
      >
        Time To Take: 
      </label>
      <input
        className="hidden-text-box left-padding font-small"
        name="time_to_take"
        onChange={[Function]}
        required={true}
        type="time"
        value="12:34"
      />
    </div>
    <div
      className="add-med-input font-small quantity-input"
    >
      <label
        className="left-padding"
      >
        Quantity: 
      </label>
      <input
        className="hidden-text-box left-padding font-small"
        name="quantity"
        onChange={[Function]}
        required={true}
        type="number"
        value="2"
      />
    </div>
    <svg
      className="time-to-take-button"
      fill="currentColor"
      height="1em"
      onClick={[Function]}
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      version="1.1"
      viewBox="0 0 16 16"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 6.5v3c0 0.276 0.224 0.5 0.5 0.5h15c0.276 0 0.5-0.224 0.5-0.5v-3c0-0.276-0.224-0.5-0.5-0.5h-15c-0.276 0-0.5 0.224-0.5 0.5z"
      />
    </svg>
  </div>
  <div
    className="med-list-time-to-take-quantity-group"
  >
    <div
      className="add-med-input font-small undefined"
    >
      <label
        className="left-padding"
      >
        Time To Take: 
      </label>
      <input
        className="hidden-text-box left-padding font-small"
        name="time_to_take"
        onChange={[Function]}
        required={true}
        type="time"
        value="01:00"
      />
    </div>
    <div
      className="add-med-input font-small quantity-input"
    >
      <label
        className="left-padding"
      >
        Quantity: 
      </label>
      <input
        className="hidden-text-box left-padding font-small"
        name="quantity"
        onChange={[Function]}
        required={true}
        type="number"
        value="1"
      />
    </div>
    <svg
      className="time-to-take-button"
      data-testid="add-time-to-take-button"
      fill="currentColor"
      height="1em"
      onClick={[Function]}
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      version="1.1"
      viewBox="0 0 16 16"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M15.5 6h-5.5v-5.5c0-0.276-0.224-0.5-0.5-0.5h-3c-0.276 0-0.5 0.224-0.5 0.5v5.5h-5.5c-0.276 0-0.5 0.224-0.5 0.5v3c0 0.276 0.224 0.5 0.5 0.5h5.5v5.5c0 0.276 0.224 0.5 0.5 0.5h3c0.276 0 0.5-0.224 0.5-0.5v-5.5h5.5c0.276 0 0.5-0.224 0.5-0.5v-3c0-0.276-0.224-0.5-0.5-0.5z"
      />
    </svg>
  </div>
  <div
    className="add-med-input font-small"
  >
    <input
      className="hidden-submit-button font-small"
      type="submit"
      value="Add medication"
    />
  </div>
</form>
`;

exports[`check no unintended changes have been made to rendered component 3`] = `
<form
  className="add-med-form"
  onSubmit={[Function]}
>
  <div
    className="add-med-input font-small undefined"
  >
    <label
      className="left-padding"
    >
      Name: 
    </label>
    <input
      className="hidden-text-box left-padding font-small"
      name="med_name"
      onChange={[Function]}
      required={true}
      type="text"
      value="2hwqlkmgtnqpewjirn"
    />
  </div>
  <div
    className="add-med-input font-small undefined"
  >
    <label
      className="left-padding"
    >
      Start Date: 
    </label>
    <input
      className="hidden-text-box left-padding font-small"
      name="start_date"
      onChange={[Function]}
      required={true}
      type="date"
      value="2021-11-20"
    />
  </div>
  <div
    className="add-med-input font-small undefined"
  >
    <label
      className="left-padding"
    >
      End Date: 
    </label>
    <input
      className="hidden-text-box left-padding font-small"
      name="end_date"
      onChange={[Function]}
      required={true}
      type="date"
      value="2021-11-29"
    />
  </div>
  <div
    className="med-list-time-to-take-quantity-group"
  >
    <div
      className="add-med-input font-small undefined"
    >
      <label
        className="left-padding"
      >
        Time To Take: 
      </label>
      <input
        className="hidden-text-box left-padding font-small"
        name="time_to_take"
        onChange={[Function]}
        required={true}
        type="time"
        value="12:34"
      />
    </div>
    <div
      className="add-med-input font-small quantity-input"
    >
      <label
        className="left-padding"
      >
        Quantity: 
      </label>
      <input
        className="hidden-text-box left-padding font-small"
        name="quantity"
        onChange={[Function]}
        required={true}
        type="number"
        value="1"
      />
    </div>
    <svg
      className="time-to-take-button"
      fill="currentColor"
      height="1em"
      onClick={[Function]}
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      version="1.1"
      viewBox="0 0 16 16"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 6.5v3c0 0.276 0.224 0.5 0.5 0.5h15c0.276 0 0.5-0.224 0.5-0.5v-3c0-0.276-0.224-0.5-0.5-0.5h-15c-0.276 0-0.5 0.224-0.5 0.5z"
      />
    </svg>
  </div>
  <div
    className="med-list-time-to-take-quantity-group"
  >
    <div
      className="add-med-input font-small undefined"
    >
      <label
        className="left-padding"
      >
        Time To Take: 
      </label>
      <input
        className="hidden-text-box left-padding font-small"
        name="time_to_take"
        onChange={[Function]}
        required={true}
        type="time"
        value="03:00"
      />
    </div>
    <div
      className="add-med-input font-small quantity-input"
    >
      <label
        className="left-padding"
      >
        Quantity: 
      </label>
      <input
        className="hidden-text-box left-padding font-small"
        name="quantity"
        onChange={[Function]}
        required={true}
        type="number"
        value="2"
      />
    </div>
    <svg
      className="time-to-take-button"
      fill="currentColor"
      height="1em"
      onClick={[Function]}
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      version="1.1"
      viewBox="0 0 16 16"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 6.5v3c0 0.276 0.224 0.5 0.5 0.5h15c0.276 0 0.5-0.224 0.5-0.5v-3c0-0.276-0.224-0.5-0.5-0.5h-15c-0.276 0-0.5 0.224-0.5 0.5z"
      />
    </svg>
  </div>
  <div
    className="med-list-time-to-take-quantity-group"
  >
    <div
      className="add-med-input font-small undefined"
    >
      <label
        className="left-padding"
      >
        Time To Take: 
      </label>
      <input
        className="hidden-text-box left-padding font-small"
        name="time_to_take"
        onChange={[Function]}
        required={true}
        type="time"
        value="04:00"
      />
    </div>
    <div
      className="add-med-input font-small quantity-input"
    >
      <label
        className="left-padding"
      >
        Quantity: 
      </label>
      <input
        className="hidden-text-box left-padding font-small"
        name="quantity"
        onChange={[Function]}
        required={true}
        type="number"
        value="3"
      />
    </div>
    <svg
      className="time-to-take-button"
      fill="currentColor"
      height="1em"
      onClick={[Function]}
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      version="1.1"
      viewBox="0 0 16 16"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 6.5v3c0 0.276 0.224 0.5 0.5 0.5h15c0.276 0 0.5-0.224 0.5-0.5v-3c0-0.276-0.224-0.5-0.5-0.5h-15c-0.276 0-0.5 0.224-0.5 0.5z"
      />
    </svg>
  </div>
  <div
    className="med-list-time-to-take-quantity-group"
  >
    <div
      className="add-med-input font-small undefined"
    >
      <label
        className="left-padding"
      >
        Time To Take: 
      </label>
      <input
        className="hidden-text-box left-padding font-small"
        name="time_to_take"
        onChange={[Function]}
        required={true}
        type="time"
        value="05:00"
      />
    </div>
    <div
      className="add-med-input font-small quantity-input"
    >
      <label
        className="left-padding"
      >
        Quantity: 
      </label>
      <input
        className="hidden-text-box left-padding font-small"
        name="quantity"
        onChange={[Function]}
        required={true}
        type="number"
        value="123"
      />
    </div>
  </div>
  <div
    className="add-med-input font-small"
  >
    <input
      className="hidden-submit-button font-small"
      type="submit"
      value="edit"
    />
  </div>
</form>
`;
